### PR TITLE
Move to Python 3.6+ and integration with new version of neo-boa

### PR DESCRIPTION
**Pull request by localhuman**
**What current issue(s) does this address, or what feature is it adding?**

- move to only python 3.6+ support
- use new version of compiler ( neo-boa==0.3.3 ) based on python 3.6 wordcode
- full testing of VM and all compiled smart contracts
- adds new command `TestBuild` for running tests of compiled contracts

**How did you solve this problem?**
- wrote new version of compiler for python 3.6+
- added a lot of tests
- updated travis

**How did you make sure your solution works?**
- tests

**Are there any special changes in the code that we should be aware of?**
- This is not compatible with Python 3.5.  You will need to delete your virtual environment and create a new one with Python 3.6 or greater, then re-install `requirements.txt`

**Please check the following, if applicable:**

- [x] Did you add any tests?
- [x] Did you run `make lint`?
- [x] Did you run `make test`?
- [x] Are you making a PR to a feature branch or development rather than master?
- [ ] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

File: requirements_docs.txt [hunk 1]
```diff
@@ -38,6 +38,7 @@ mock==2.0.0
 mpmath==1.0.0
 neo-boa==0.2.1
 neo-python-rpc==0.1.8
+neo-core==0.3.6
```
Comment (metachris) [inline requirements_docs.txt]: `neo-core` doesn't exist, the package is called `neocore`. Why do we need this in the docs requirements, actually?

File: requirements_docs.txt [hunk 2]
```diff
@@ -9,7 +9,7 @@ base58==0.2.5
 bitcoin==1.1.42
 blessings==1.6.1
 bpython==0.17
-byteplay3==3.5.0
+bytecode==0.5
```
Comment (metachris) [inline requirements_docs.txt]: Why do we need this in the docs requirements?

**Comment by metachris** [comment]
If we merge this into master, we should definitely do a minor version bump!

File: requirements_docs.txt (hunk 2 above)
Comment (localhuman) [reply requirements_docs.txt]: I think at some point we would maybe be using the compiler when generating the docs?

Comment (metachris) [reply requirements_docs.txt]: Not sure about the use-case, but sure :D

**Comment by metachris** [comment]
woo hoo

**Pull request merged**